src/main/java/org/gradle/playframework/tools/internal/less/Less4jCompiler.java

```diff
@@ -0,0 +1,108 @@
+
+            didWork |= compile(lessFile.getFile(), cssFile, includePaths);
+        }
+        return WorkResults.didWork(didWork);
+    }
+
+    private boolean compile(File lessFile, File cssFile, File[] includePaths) {
+        loadCompilerClasses(getClass().getClassLoader());
+
+        Object lessSource = DirectInstantiator.INSTANCE.newInstance(multiSourceClass, lessFile, includePaths);
+        Object options = DirectInstantiator.INSTANCE.newInstance(configurationClass);
+
+        JavaMethod<Object, Void> setCssResultLocation = JavaReflectionUtil.method(configurationClass, Void.class, "setCssResultLocation", File.class);
+        setCssResultLocation.invoke(options, cssFile);
+
+        JavaMethod<Object, Object> setCompressing = JavaReflectionUtil.method(configurationClass, configurationClass, "setCompressing", boolean.class);
+        setCompressing.invoke(options, true);
+
+        Object compiler = DirectInstantiator.INSTANCE.newInstance(compilerClass);
+
+        JavaMethod<Object, Object> doCompile = JavaReflectionUtil.method(compilerClass, Object.class, "compile", sourceClass, configurationClass);
+        Object result = doCompile.invoke(compiler, lessSource, options);
+
+        JavaMethod<Object, String> getCss = JavaReflectionUtil.method(resultClass, String.class, "getCss");
+        String css = getCss.invoke(result);
+
+        GFileUtils.writeFile(css, cssFile);
+        return true;
+    }
+
+    private void loadCompilerClasses(ClassLoader cl) {
+        try {
+            if (sourceClass == null) {
+                @SuppressWarnings("unchecked")
+                Class<Object> clazz = (Class<Object>) cl.loadClass("com.github.sommeri.less4j.LessSource");
+                sourceClass = clazz;
+            }
+            if (multiSourceClass == null) {
+                @SuppressWarnings("unchecked")
+                Class<Object> clazz = (Class<Object>) cl.loadClass("com.github.sommeri.less4j.MultiPathFileSource");
+                multiSourceClass = clazz;
+            }
+            if (configurationClass == null) {
+                @SuppressWarnings("unchecked")
+                Class<Object> clazz = (Class<Object>) cl.loadClass("com.github.sommeri.less4j.LessCompiler$Configuration");
+                configurationClass = clazz;
+            }
+            if (resultClass == null) {
+                @SuppressWarnings("unchecked")
+                Class<Object> clazz = (Class<Object>) cl.loadClass("com.github.sommeri.less4j.LessCompiler$CompilationResult");
+                resultClass = clazz;
+            }
+            if (compilerClass == null) {
+                @SuppressWarnings("unchecked")
+                Class<Object> clazz = (Class<Object>) cl.loadClass("com.github.sommeri.less4j.core.ThreadUnsafeLessCompiler");
+                compilerClass = clazz;
+            }
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
```

Comment:
Providing more context about what's going on here would be useful.

Reply:
Added context!